### PR TITLE
Remove SnapshotStoreSupplier

### LINKED ISSUE #7409: Remove dependency to SnapshotStoreSupplier in partitions
Currently the dependency to `SnapShotStoreSupplier` is a bit unwieldy. The supplier is created very early and globally.
After refactoring the bootstrapping, the snapshot store will be created alongside the partition.

As an intermediate step, the goal of this issue is to remove the dependency to `SnapshotStoreSupplier` from the partition. This class is only used in the transition. It is used in several ways:

`ConstructableSnapshotStore getConstructableSnapshotStore(int partitionId);` and `ReceivableSnapshotStore getReceivableSnapshotStore(final int partitionId);` are passed on into `StateControllerImpl`

The method `PersistedSnapshotStore getPersistedSnapshotStore(int partitionId);` is used by several callers to eventually register a snapshot listener.

Therefore, to reduce the dependencies:
* the method `PartitionTransitionContext#getSnapshotStoreSupplier()` will be removed.
* two new methods will be added to `PartitionTransitionContext`
  * `getConstructableSnapshotStore()`
  * `getReceivableSnapshotStore()`
*  `LogDeletionService` will use `getReceivableSmapshotStore()` _and_ `getConstructableSmapshotStore()` instead to register itself as listener (it's currently the same object, but there is a set, so it should be fine. If these ever are two objects, my thought is that we want to compact the log regardless of who created the snapshot)
* `LeaderPostStoragePartitionStep` and `ZeebeDbPartitionStep` will use `getConstructableSnapshotStore()` instead to register itself as listener (with the rationale, that in this case the newly created snapshot shall be replicated)

### PR DESCRIPTION
**Pull request by pihme**
## Description
This removes the `SnapshotStoreSupplier` interface. The partition components now directly depend in the snapshot interfaces.

## Related issues

closes #7409

## Definition of Done

Code changes:
* [X] The changes are backwards compatibility with previous versions
* [ ] If it fixes a bug then PRs are created to [backport](https://github.com/zeebe-io/zeebe/compare/stable/0.24...develop?expand=1&template=backport_template.md&title=[Backport%200.24]) the fix to the last two minor versions. You can trigger a backport by assigning labels (e.g. `backport stable/0.25`) to the PR, in case that fails you need to create backports manually.

Testing:
* [ ] There are unit/integration tests that verify all acceptance criterias of the issue
* [ ] New tests are written to ensure backwards compatibility with further versions
* [ ] The behavior is tested manually
* [ ] The change has been verified by a QA run
* [ ] The impact of the changes is verified by a benchmark 

Documentation: 
* [ ] The documentation is updated (e.g. BPMN reference, configuration, examples, get-started guides, etc.)
* [ ] New content is added to the [release announcement](https://drive.google.com/drive/u/0/folders/1DTIeswnEEq-NggJ25rm2BsDjcCQpDape)

### CODE REVIEW

**Comment by pihme** [comment]
Will do a rebase to pull in changes of other PRs and remove merge conflict

File: broker/src/main/java/io/camunda/zeebe/broker/logstreams/LogDeletionService.java [hunk 1]
```diff
@@ -35,14 +36,12 @@ public String getName() {
 
   @Override
   protected void onActorStarting() {
-    persistedSnapshotStore.addSnapshotListener(this);
```
Comment (Zelldon) [inline broker/src/main/java/io/camunda/zeebe/broker/logstreams/LogDeletionService.java:38]: Wouldn't this mean we are triggered twice now?

**Review by ChrisKujawa — COMMENTED** [review]
Thanks @pihme I'm just a bit unsure about the LogDeletionService

File: broker/src/main/java/io/camunda/zeebe/broker/logstreams/LogDeletionService.java (hunk 1 above)
Comment (pihme) [reply broker/src/main/java/io/camunda/zeebe/broker/logstreams/LogDeletionService.java:38]: No, the snapshot store uses a set, so we will be triggered once.

Comment (Zelldon) [reply broker/src/main/java/io/camunda/zeebe/broker/logstreams/LogDeletionService.java:38]: thanks

**Review by ChrisKujawa — APPROVED** [review]

**Pull request merged**